src/classes/issues-processor.ts

```diff
@@ -556,6 +557,7 @@ export class IssuesProcessor {
         (issue: Readonly<OctokitIssue>): Issue => new Issue(this.options, issue)
       );
     } catch (error) {
+      this.errors.push(error);
       this._logger.error(`Get issues for repo error: ${error.message}`);
       return Promise.resolve([]);
```

Comment:
```suggestion
      this.errors.push(error);
      this._logger.error(`Get issues for repo error: ${error.message}`);
      core.setFailed(error.message);
```
In my opinion, if there is a problem with fetching data from the repository, we can simply fail the action. Also it will eliminate the need of any other code that was added to handle errors in this file and in main.ts. Also we can try to add retry logic for fetching, for that reason we need to take a look at Octokit plugin https://github.com/octokit/plugin-retry.js, it will retry fetching several times in case of errors with codes 4xx and 5xx. We can enable it in the line 78 by adding it as the argument to the function `getOctokit(this.options.repoToken);` Check the signature of the `getOctokit()` https://github.com/actions/toolkit/blob/main/packages/github/src/github.ts